advertools/spider.py

```diff
@@ -939,10 +968,13 @@ def parse(self, response):
                     for norm in ld_norm:
                         jsonld.update(**norm)
         except Exception as e:
-            jsonld = {"jsonld_errors": str(e)}
-            self.logger.exception(
-                " ".join([str(e), str(response.status), response.url])
-            )
+            if 'jsonld_errors' not in self.discard_columns: 
```

Comment:
Can you please explain why we are adding this condition?
I think logging jsonld_errors is always good, even if it's in the discard_columns, this is just logging to the terminal, optionally to a specific file.

The important thing is that adding `jsonld_errors` to `discard_columns` ensures that it's not in the output_file. 

Unless I'm missing something?

Reply:
@eliasdabbas, my objetive was to fix the OOM error that started this PR. In our case, the OOM was caused by a part of the code that is now within the if statement.

That being said, you're right and we should print it. Maybe we should add only this part within the condition ? 

```python 
jsonld = {"jsonld_errors": str(e)}
```

The error would still be printed but not stored in a variable; hence the OOM wouldn't happen. 

What do you think?

Reply:
@antoineeripret

I'm not sure why you should get an OOM error, unless you are running this in a jupyter notebook? 
In this case logs can kill the browser's memory, yes. Logs can be redirected to a file (recommended), or disabled.

`discard_columns` means don't include them in the output file, but deciding to implicitly not log those errors without the user knowing might be misleading, especially in this specific case where you have a massive amount of errors, and you should know about them.